### ISSUE TITLE
TensorPrimitives.IndexOfXx methods sometimes return the wrong index in the face of equality

### ISSUE THREAD

**Issue by stephentoub** [dotnet/runtime#97190]
### Description

When there are multiple equal elements that could have their index returned, the one with the smallest index is supposed to be returned. But it isn't always.

### Reproduction Steps

```C#
using System.Numerics.Tensors;

ReadOnlySpan<float> floats = [10f, -20f, -30f, 1f, 1f, -40f];
Console.WriteLine(TensorPrimitives.IndexOfMinMagnitude(floats));
```
or
```C#
using System.Numerics.Tensors;

ReadOnlySpan<float> floats = [10f, 20f, 30f, 1f, 1f, 40f];
Console.WriteLine(TensorPrimitives.IndexOfMin(floats));
```

### Expected behavior

Those should both print `3`.

### Actual behavior

They print `4`.

### Regression?

No

### Configuration

System.Numerics.Tensors v8.0.0 nuget package

**Comment by tannergooding** [comment]
I think the bug is here in the equality handling check: https://source.dot.net/#System.Numerics.Tensors/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs,10125

We'll get down to the `MinAcross` logic and end up comparing `result = [10, 1, 1, 1]; resultIndex = [0, 3, 4, 3]` to `tmpResult = [1, 1, 10, 1]; tmpIndex = [4, 3, 0, 3]`.

The `lessThanMask` will be `[false, false, true, false]`, the `equalMask` will be `[false, true, false, true]`, the `negativeMask` will be `[false, false, false, false]`, and the `lessThanIndexMask` will be `[true, false, false, false]`.

We will then modify the `lessThanMask` using the other masks, giving us `[false, true, true, true]`. This then picks `result = [1, 1, 1, 1]; resultIndex = [4, 3, 4, 3]` and will compare that to `tmpResult = [1, 1, 1, 1]; tmpIndex = [3, 4, 3, 4]`.

The `lessThanMask` will be `[false, false, false, false]`, the `equalMask` will be `[true, true, true, true]`, the `negativeMask` will be `[false, false, false, false]`, and the `lessThanIndexMask` will be `[false, true, false, true]`.

We will then modify the `lessThanMask` using the other masks, giving us `[true, true, true, true]` which causes us to preserve `result = [1, 1, 1, 1]; resultIndex = [4, 3, 4, 3]` and return `4` since its the first entry.

---------------------------------

So the failure here looks to come about because `lessThanIndexMask` is only used when there is some negative in `result`. We'd need to modify this to use it in more than just this scenario to ensure that the lesser index is picked.

What we have now is effectively:
```
Vector128<float> currentNegativeMask = IsNegative(current);

// Take the entry from current if it is equal to result and negative
lessThanMask |= Vector128.AndNot(equalMask, currentNegativeMask);

// Take the entry from result if it is equal to current and negative and its index is lesser
lessThanMask |= equalMask & IsNegative(result) & lessThanIndexMask;
```

What we need to say is:
* If both are equal and have the same sign, pick the lesser index
* If both are equal and only one is negative, pick the negative one

This is required to ensure `+0` vs `-0` is handled correctly.

**Comment by stephentoub** [comment]
Thanks for the detailed look. I actually already have a fix included in #97192 (though I neglected to copy the fix over to the netstandard impl, I'm doing that now). I basically ended up with this in IndexOfMax (and similar in the others):
```C#
                    Vector128<T> lessThanIndexMask = IndexLessThan(resultIndex, curIndex);
                    if (typeof(T) == typeof(float) || typeof(T) == typeof(double))
                    {
                        greaterThanMask |= (equalMask & (~IsNegative(result) | IsNegative(current)) & lessThanIndexMask);
                    }
                    else
                    {
                        greaterThanMask |= (equalMask & lessThanIndexMask);
                    }
```
